<!--When used by exo call this after launching march.launch, when using test joint use the correct march_type-->
<launch>
    <arg name="configuration" default="None"/>
    <arg name="linear" default="true"/>
    <arg name="slope" default="10.0"/>

    <group if="$(eval configuration == 'exoskeleton')">
        <include file="$(find march_launch)/launch/march.launch">
            <arg name="configuration" value="$(arg configuration)"/>
        </include>
    </group>

    <group if="$(eval configuration == 'simulation')">
        <include file="$(find march_launch)/launch/march.launch">
            <arg name="configuration" value="$(arg configuration)"/>
        </include>
    </group>

    <group if="$(eval configuration == 'test_joint_rotational')">
        <include file="$(find march_launch)/launch/march.launch">
            <!--This value is not used in the launch file and therefore allows us to start the test joint ourselves.-->
            <arg name="configuration" value="$(arg configuration)"/>
        </include>
        <include file="$(find march_hardware_interface)/launch/hardware.launch">
            <arg name="robot" value="$(arg configuration)"/>
        </include>
    </group>

    <node name="gain_scheduling_node" pkg="march_gain_scheduling" type="march_gain_scheduling_node" output="screen">
        <rosparam command="load" file="$(find march_gain_scheduling)/config/$(arg configuration)/gains_per_gait_type.yaml"/>
        <param name="linearize_gain_scheduling" type="bool" value="$(arg linear)"/>
        <param name="linear_slope" type="double" value="$(arg slope)"/>
    </node>
</launch>

